server/src/main/java/com/cloud/api/ApiResponseHelper.java

```diff
@@ -2719,6 +2750,29 @@ public NetworkResponse createNetworkResponse(ResponseView view, Network network)
         response.setBytesReceived(bytesReceived);
         response.setBytesSent(bytesSent);
 
+        if (networkOfferingDao.isRoutedNetwork(network.getNetworkOfferingId())) {
```

Comment:
@shwstppr 
indeed this could be optimized.

However, this is consistent as what the ipv6routes are set.
also, in almost all users's environments, there are very few (less than 100) network offerings, as numbers and bgp peers (user_ip_address table might be big), so it may be not a major issue if there are lots of networks.

The most time of listnetworks API occurs in the query of `networks` table which could be very big for long-running environments, not in the creation of response (except `user_statistics` table which could be big).

Reply:
@shwstppr 
I had another look at thes issue in the search of networks (#9184 aimed to fix) and vms (#8012 aimed to fix)
it looks both issues are related to the query on the tables (not the creation of response).
Also `view` do not help sometimes, if there are several join tables (see #8012).